src/analysisd/decoders/security_configuration_assessment.c

```diff
@@ -926,14 +968,21 @@ static void HandleScanInfo(Eventinfo *lf,int *socket,cJSON *event) {
                 description_db = description->valuestring;
             }
 
-            result_event = SavePolicyInfo(lf,socket,policy->valuestring,file->valuestring,policy_id->valuestring,description_db,references_db);
+            result_event = SavePolicyInfo(lf,socket,policy->valuestring,file->valuestring,policy_id->valuestring,description_db,references_db,hash_file->valuestring);
             if (result_event < 0)
             {
                 merror("Error storing scan policy monitoring information for agent %s", lf->agent_id);
             }
-            
             break;
         default:
+            os_calloc(OS_MAXSTR, sizeof(char), old_hash);
+            if(FindPolicySHA256(lf, policy_id->valuestring, socket, old_hash) == 0){
+                if(strcmp(hash_file->valuestring, old_hash)){
+                    DeletePolicyCheck(lf, policy_id->valuestring, socket);
```

Comment:
When deleting the policy, we are not controlling the possible errors as it is done here:
https://github.com/wazuh/wazuh/blob/99cfc0dc01bcd277b2e78b1841e6c2f768ee9616/src/analysisd/decoders/security_configuration_assessment.c#L1240

In addition, delete the policy information before its corresponding checks.